src/library/scala/collection/SeqLike.scala

```diff
@@ -232,7 +232,7 @@ trait SeqLike[+A, +Repr] extends Any with IterableLike[A, Repr] with GenSeqLike[
       if (idx < 0)
         _hasNext = false
       else {
-        var sum = nums.slice(idx + 1, nums.length).sum + 1
+        var sum = nums.view(idx + 1, nums.length).sum + 1
```

Comment:
This is still considerably slower than a direct computation.  If it matters at all, it should be switched to an accumulator.